subprojects/build-operations/src/main/java/org/gradle/trace/buildops/BuildOperationTrace.java

```diff
@@ -46,6 +57,51 @@ public void started(BuildOperationDescriptor buildOperationDescriptor, Operation
                 writer.finished();
             }
         }
+    }
+
+    private static class BuildOperationDurationRecordingListener extends DetachingBuildOperationListener {
```

Comment:
> `extends DetachingBuildOperationListener`

Doesn't this mean that the listener detaches before the first task would ever execute?

Reply:
The listener detaches after all the tasks finished executing (in finished).

Reply:
Ah, right, missed that.